lib/Di.php

```diff
@@ -0,0 +1,237 @@
+<?php
```

Comment:
Agreed that this lib definitely needs a DI given the number of files that using the Provider/Factory pattern requires, not a required change for merge, but would generally just use an existing DI like Symfony DI or similar. But happy that this more than gets the job done